tests/res_diff_expect/test_revisions_with_diff_list.ion

```diff
@@ -0,0 +1,17 @@
+$ion_1_0
+{
+ test: {
+  read_compare: {
+   message: "Read_compare: two revisions agree with each other but have different disagree lists. This might be a cli tool issue.",
```

Comment:
Again, I would remove `This might be a cli tool issue.` from everywhere. It could be a problem in the CLI or the implementation itself; either way, it should be fixed.

In this case, since the implementations that the latter revision disagrees with is a subset of the implementations that the former revision disagrees with, should we print this difference but not fail the `--results-diff` invocation due to this?

Reply:
I think two revisions may be supposed to have the same disagree list as long as they agree with each other.

For example, if impl_1 agrees with impl_2 and impl_2 disagrees with impl_3. impl_1 has to disagree with impl_3, right? Otherwise It's an issue.